plugin/pkg/scheduler/algorithm/priorities/priorities_test.go

```diff
@@ -52,7 +55,26 @@ func makeNode(node string, milliCPU, memory int64) *api.Node {
 	}
 }
 
-func TestZeroRequest(t *testing.T) {
+func priorityFunction(mapFn algorithm.PriorityMapFunction, reduceFn algorithm.PriorityReduceFunction) algorithm.PriorityFunction {
+	return func(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodes []*api.Node) (schedulerapi.HostPriorityList, error) {
+		result := make(schedulerapi.HostPriorityList, 0, len(nodes))
+		for i := range nodes {
+			hostResult, err := mapFn(pod, nil, nodeNameToInfo[nodes[i].Name])
+			if err != nil {
+				return nil, err
+			}
+			result = append(result, hostResult)
+		}
+		if reduceFn != nil {
+			if err := reduceFn(result); err != nil {
+				return nil, err
+			}
+		}
+		return result, nil
+	}
+}
+
+/*func TestZeroRequest(t *testing.T) {
```

Comment:
I think that fixing this test, is the only missing part of this PR for now.

In general, changing all priority functions to the new signature may require some small changes, but I took a look into e.g. SelectorSpreading (which seems to be the most problematic), and it should fit this framework.